core/commands/get.go

```diff
@@ -66,33 +66,30 @@ may also specify the level of compression by specifying '-l=<1-9>'.
 			return err
 		}
 
-		node, err := cmdenv.GetNode(env)
+		api, err := cmdenv.GetApi(env, req)
 		if err != nil {
 			return err
 		}
-		p := path.Path(req.Arguments[0])
-		ctx := req.Context
-		dn, err := core.Resolve(ctx, node.Namesys, node.Resolver, p)
+
+		p, err := iface.ParsePath(req.Arguments[0])
 		if err != nil {
 			return err
 		}
 
-		switch dn := dn.(type) {
-		case *dag.ProtoNode:
-			size, err := dn.Size()
-			if err != nil {
-				return err
-			}
+		file, err := api.Unixfs().Get(req.Context, p)
+		if err != nil {
+			return err
+		}
 
-			res.SetLength(size)
-		case *dag.RawNode:
-			res.SetLength(uint64(len(dn.RawData())))
-		default:
+		size, err := file.Size()
+		if err != nil {
 			return err
 		}
 
+		res.SetLength(uint64(size))
```

Comment:
This is actually incorrect but it's an existing bug... https://github.com/ipfs/go-ipfs/issues/5690

Reply:
The alternative is not setting this at all, but it will break progress bar... (We could pass this in a separate header)